### ISSUE TITLE
ENOENT: no such file or directory, rmdir node_modules/.vite/temp

### ISSUE THREAD

**Issue by danielroe** [nuxt/vite#2]
```
ERROR  Client compiled with errors: ENOENT: no such file or directory, rmdir '/my/repo/node_modules/.vite/temp'
```

**Comment by pi0** [comment]
This is a race condition which couldn't figure out how to solve... Ideally we should be able to move vite's temp directory out of buildDir since builder cleans this directory

**Comment by pi0** [comment]
It seems we still have this issue even by moving `.vite` out of `.nuxt`

**Comment by antfu** [comment]
How about using `ensureDir` for `mkdirp`?

**Comment by pi0** [comment]
@antfu actually we are doing it (https://github.com/nuxt/vite/commit/2f2cd4c3b3e0bbf804af8b3da72da65bf0d59de4#diff-46581ec84ce8464d8a2624f7e2c556ad1a27f072f813b8056c2344da67d43b75R49). Is suspected maybe because two vite builds are starting there is a race condition between

**Comment by pi0** [comment]
Cache fixed by ff92525f41f59a72979f28580c67741a3d841fc8 but now vite is broken with an import :}

**Comment by pi0** [comment]
Should be fixed in 0.0.8+

causes:
 - First cause was using `.nuxt` directory which nuxt recreates it on each template regen
 - Second cause was using same cacheDir for client and server
